worker-workflow/src/main/java/com/github/cafdataprocessing/workflow/FailureFieldsManager.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2015-2018 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafdataprocessing.workflow;
+
+import com.google.gson.Gson;
+import com.hpe.caf.worker.document.model.Document;
+import java.util.HashMap;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class FailureFieldsManager
+{
+    private static final Logger LOG = LoggerFactory.getLogger(FailureFieldsManager.class);
+    private final Gson gson;
+
+    public FailureFieldsManager()
+    {
+        this.gson = new Gson();
+    }
+
+    public void handleExtraFailureSubFields(final Document document)
+    {
+        if (!PoisonMessageDetector.isPoisonDocument(document)) {
+            final Map<String, String> extraFailureSubfields = retrieveExtraFailureSubfields(document);
+            if (!extraFailureSubfields.isEmpty()) {
```

Comment:
I'm still not convinced that this can't be spoofed.

Reply:
Updated as discussed to prevent spoofing of failure fields